Add release news for Shopware v6.5.7.0 (#1)
shopware/release-notes: src/6.5.7.x/6.5.7.0.md

> # Release notes Shopware 6.5.7.0

Author: Right, they call it like that: https://www.shopware.com/en/news/shopware-6-release-news-october-2023/. But, to be very honest: as a developer, I am more interested in the version number, don't care the season when it was published.

> # Release notes Shopware 6.5.7.0

Reviewer: I'm with Marco on this one. On top of that, we have multiple releases every month - https://www.shopware.com/en/changelog/

> ## Abstract

Author: It shall rather be an abstract than an introduction a la "this patch release contains two bug fixes, we didn't touch the storefront at all". We can also call it _synopsis_ if _abstract_ is misleading.

### Media Path refactoring

We refactored the media path handling to persist the media paths instead of recalculate in any request. This allows that media entity can be queried only with the path. Previously the complete media entity needed to be loaded, to generate the path.

```http
POST /api/search/product

{
    "fields": [
        "name",
        "cover.path"
    ]
}
```

### Improvments to logging

As in the previous releases, we continued to improve the logging of Shopware. Hence, we added more logging or down-ranked non-relevant errors to noises to keep the logs clean for important messages.

### Deprecated LoggerFactory

We deprecated the LoggerFactory for plugins for 6.6 as this does not consider project configuration of logging. Our suggested replacement is to use Monolog Channels, which works in any Shopware 6 version. Please also see the PayPal plugin as an example of our implementation:

- https://github.com/shopware/SwagPayPal/blob/7.3.0/src/Resources/config/packages/monolog.yaml
- https://github.com/shopware/SwagPayPal/blob/7.3.0/src/SwagPayPal.php#L200-L217

### New low_priority queue

> We introduced a new queue `low_priority` additional to the already existing `async` queue. This queue can be used by plugins, to queue up messages which are not important in a short manner.

Reviewer 2: ```suggestion
We introduced a new queue, `low_priority`, in addition to the already existing `async` queue. This queue can be used by plugins, to queue up messages which are not important in a short manner.
```

Reviewer 2: What does in a short manner.

The existing `messenger:consume` or admin watcher will consume the new queue without configuration changes. With Shopware 6.6, it becomes required to setup an additional `messenger:consume` or just append the queue additionally to the existing one `messenger:consume async,low_priority`

### Improved plugin:create

> The `plugin:create` command now can generate also:

Reviewer 2: ```suggestion
The `plugin:create` command now can generate:
```

- console command
- scheduled task
- event subscriber
- storefront controller
- store-api route

We are working right now also on a `dal:migration:create` to generate migration files automatically by comparing your entity definition with the current database schema. So stay tuned!

### Soft delete of custom entities on App uninstall

Custom Entities created by Apps will now be soft deleted, so that you can preserve the app data on uninstallation.

### System update now uses the regular kernel

> The commands `system:update:prepare` and `system:update:finish` now use the regular kernel while updating Shopware. This fixes a lot of issues when plugins extended root components like Symfony Configuration or created themes.

Reviewer 2: ```suggestion
The commands `system:update:prepare` and `system:update:finish` now use the regular kernel while updating Shopware. This fixes a lot of issues when plugins extend root components like Symfony Configuration or created themes.
```

### MySQL 8.2 support

We fixed some internal queries of DAL to be able to support MySQL 8.2.

### And many more things

* Paginated pages in listing are now HTTP cached
* Improved performance *order_line_item* migration
* Removed *ManyToManyIdFieldUpdater* in *LandingPageIndexer*
* Optimized SEO lookup queries

## Fixed bugs

[TBD]

* important and frequently requested bugfixes go here with a linked NEXT-number as well as a short description/headline
* and another important one peeps longing for
* and yet another one

See all bugs fixed with this release: [link to a list of all bugs fixed with this release](link to a list of all bugs fixed with this release)

## Credits

> Thanks to all thou diligent friends to help us making Shopware better and better with each pull request!

Reviewer 2: ```suggestion
Thanks to all diligent friends for helping us make Shopware better and better with each pull request!
```

[TBD]

* [maximilianruesch](https://github.com/maximilianruesch) for [NEXT-30599](https://github.com/shopware/shopware/blob/v6.5.6.0/changelog/release-6-5-6-0/2023-08-09-fix-overridden-super-calls-in-promise-chains-in-admin-components.md) and [NEXT-30427](https://github.com/shopware/shopware/blob/v6.5.6.0/changelog/release-6-5-6-0/2023-08-14-fix-sw-datepicker-to-load-refs-correctly-on-first-page-load.md)

> ## More ressources:

Reviewer 2: ```suggestion
## More resources:
```

> * Study a detailled [diff on Github](https://github.com/shopware/shopware/compare/v6.5.7.0...v6.5.6.2) to the former version

Reviewer 2: ```suggestion
* [Detailed diff on Github](https://github.com/shopware/shopware/compare/v6.5.7.0...v6.5.6.2) to the former version
```

> * See the [changelog](https://github.com/shopware/shopware/blob/v6.5.7.0/CHANGELOG.md) for this version on GitHub

Reviewer 2: ```suggestion
* [Changelog on GitHub](https://github.com/shopware/shopware/blob/v6.5.7.0/CHANGELOG.md) for this version.
```

> * Read the marketing blog post at [https://www.shopware.com/en/news/shopware-6-release-news-november-2023/](https://www.shopware.com/en/news/shopware-6-release-news-november-2023/)

Reviewer 2: ```suggestion
* [Marketing blog post](https://www.shopware.com/en/news/shopware-6-release-news-november-2023/)
```

* [Installation overview](https://developer.shopware.com/docs/guides/installation/)
* [Update from a previous installation](https://developer.shopware.com/docs/guides/installation/template.html#update-shopware)

## Get in touch

> Discuss about decisions, bugs you might stumble upon etc in our [community slack](https://slack.shopware.com). See you there ;)

Reviewer 2: ```suggestion
Discuss about decisions, bugs you might stumble upon, etc in our [community slack](https://slack.shopware.com). See you there ;)
```
